js/iframe-coordinator/elements/frame-router.ts

```diff
@@ -21,9 +33,18 @@ class FrameRouterElement extends HTMLElement {
 
   registerClients(clients) {
     this.router = Host.embed(this, clients);
+    this.router.ports.toHost.subscribe(message => {
+      if (message.msgType == "publish") {
+        let event = new CustomEvent("publish", { detail: message.msg });
+        this.dispatchEvent(event);
+      }
+    });
+  }
 
-    window.addEventListener("message", event => {
-      this.router.ports.fromClient.send(event.data);
+  subscribe(topic: string): void {
+    this.router.ports.fromHost.send({
+      msgType: "subscribe",
+      msg: topic
     });
```

Comment:
When the host app subscribes, we send a message to the Elm code on `fromHost` to trigger the internal subscription state change.